### ISSUE TITLE
Cant connect to Asset Library TLS handshake error

### ISSUE THREAD

**Issue by divcyb** [godotengine/godot#95890]
### Tested versions

v4.3.stable

### System information

v4.3.stable.arch_linux - OpenGL API 4.6 (Core Profile) Mesa 24.1.6-arch1.1

### Issue description
edit: After reading below comment it seems an archlinux package specific bug, thanks.

Godot client can't connect to asset library.
from the log:
/usr/src/debug/mbedtls/mbedtls/library/ssl_tls13_generic.c:1651: 0x57c57f146228: Perform PSA-based ECDH/FFDH computation.
/usr/src/debug/mbedtls/mbedtls/library/ssl_tls13_generic.c:1689: 0x57c57f146228: psa_generate_key() returned -27648 (-0x6c00)
ERROR: TLS handshake error: -27648
   at: _do_handshake (modules/mbedtls/stream_peer_mbedtls.cpp:89)
mbedtls error: returned -0x6c00

### Steps to reproduce

Trying to connect to asset library

### Minimal reproduction project (MRP)

![20240821_12h02m24s_grim](https://github.com/user-attachments/assets/e4238636-b21e-46ea-9f10-24fa44d565a7)

similar issue:
https://github.com/godotengine/godot/issues/85953

**Comment by pixelmagickitty** [comment]
I can confirm and reproduce the **TLS handshake error**, related to loading the Asset Library, on my two Artix Linux workstations when running the **godot-4.3-2** package. Based on my tests this appears to be an issue with the **mbedtls-3.6.0-1** package. A cursory review of issues posted on the **mbedtls** repo indicates a change from 3.5.2 to 3.6.0, where TLS 1.3 was enabled by default, is the culprit. I tried a downgrade to **mbedtls-3.5.2-1** but then godot-4.3 complained about missing library versions.

I tested the official builds of...

* Godot_v4.2.2-stable_linux.x86_64
* Godot_v4.3-rc2_linux.x86_64
* Godot_v4.3-rc3_linux.x86_64
* Godot_v4.3-stable_linux.x86_64

...and the **TLS handshake error** did not occur, and the Asset Library loaded as expected. I am assuming those official builds are using the **mbedtls 3.5.2** library.

### Running the offical builds from the Godot Engine project is my current solution to this issue.


Here's the error from the terminal when running the Artix Linux **godot-4.3-2** package...

```
/build/mbedtls/src/mbedtls/library/ssl_tls13_generic.c:1651: 0x7689205e4148: Perform PSA-based ECDH/FFDH computation.
/build/mbedtls/src/mbedtls/library/ssl_tls13_generic.c:1689: 0x7689205e4148: psa_generate_key() returned -27648 (-0x6c00)
ERROR: TLS handshake error: -27648
   at: _do_handshake (modules/mbedtls/stream_peer_mbedtls.cpp:89)
mbedtls error: returned -0x6c00
```

...in case this is helpful or relavent.


Similar issue reported: [#95833](https://github.com/godotengine/godot/issues/95833), [#85953](https://github.com/godotengine/godot/issues/85953)

**Comment by christian-heusel** [comment]
Looks like https://gitlab.archlinux.org/archlinux/packaging/packages/godot/-/issues/3

**Comment by mycroftsnm** [comment]
Same issue, godot 4.3-2 on Arch

**Comment by christian-heusel** [comment]
Should be solved as per https://gitlab.archlinux.org/archlinux/packaging/packages/godot/-/issues/3#note_205130 and the following commit: https://gitlab.archlinux.org/archlinux/packaging/packages/godot/-/commit/f71396bbe8f15a517fbdae0806e0f31d5d77274f

**Comment by Calinou** [comment]
Closing per @christian-heusel's comment.

**Comment by Sammyueru** [comment]
still not fixed in 2025, Godot 4.4.1 stable.

**Comment by meisei4** [comment]
I also have this issue on arch linux 4.4.1 arch linux

**Comment by ComputerErika** [comment]
Still not fixed here as well
on v4.4.1.stable.arch_linux
Godot cant connect to the internet for asset lib, export settings, or updates
on the launcher project select it says "Failed to check for update. Error 5"
on the launcher asset lib it says "Cant connect to host: https://godotengine.org/asset-library/api"
I have no proxys enabled and havent ever used them before on this machine

Okay, Just checked the official downloaded from the website and that works and connects, maybe its a problems with my distro

**Comment by Sammyueru** [comment]
> Still not fixed here as well on v4.4.1.stable.arch_linux Godot cant connect to the internet for asset lib, export settings, or updates on the launcher project select it says "Failed to check for update. Error 5" on the launcher asset lib it says "Cant connect to host: https://godotengine.org/asset-library/api" I have no proxys enabled and havent ever used them before on this machine
> 
> Okay, Just checked the official downloaded from the website and that works and connects, maybe its a problems with my distro

Yeah I had the issue on arch as well so your probably right.

**Comment by Calinou** [comment]
Please continue the discussion in https://github.com/godotengine/godot/issues/107754, which tracks this particular issue on Arch Linux.
